test/unit/metadata.js

```diff
@@ -772,4 +773,45 @@ describe('metadata', function () {
       should(upToDate(doc).errors).be.undefined()
     })
   })
+
+  describe('generated a doc with conflict', () => {
+    it('should get the correct path', () => {
+      const doc = {
+        path: 'docname'
+      }
+      const newDoc = createConflictingDoc(doc)
+      const pathRegExp = RegExp(`(${doc.path})-conflict-\\d{4}(?:-\\d{2}){2}T(?:\\d{2}_?){3}.\\d{3}Z`)
+      should(pathRegExp.exec(newDoc.path)).not.null()
+    })
+    it('should get the correct _id', () => {
+      const doc = {
+        path: 'docname'
+      }
+      const newDoc = createConflictingDoc(doc)
+      const pathRegExp = RegExp(`(${doc.path})-conflict-\\d{4}(?:-\\d{2}){2}T(?:\\d{2}_?){3}.\\d{3}Z`)
+      should(pathRegExp.exec(newDoc._id)).not.null()
```

Comment:
Failure message could be more helpful with `should(newDoc._id).match(someRegExp)`

Reply:
Although this test will pass on Linux, `newDoc._id` will match `/-CONFLICT-/` on macOS/Windows, not `/-conflict-/`, because ids are uppercase on those platforms, because we assume users are using the corresponding default filesystems (HFS+/APFS/NTFS) which are not case sensitive.

Reply:
(see https://github.com/cozy-labs/cozy-desktop/blob/v3.11.1-beta.1/core/metadata.js#L157-L191 )